libtenzir/src/offset.cpp

```diff
@@ -57,6 +57,13 @@ auto offset::get(const arrow::RecordBatch& batch) const noexcept
 
 auto offset::get(const arrow::StructArray& struct_array) const noexcept
   -> std::shared_ptr<arrow::Array> {
+  if (empty()) {
+    // If the offset is empty, we are already at the desired level, and can just
+    // wrap the struct array back into a shared pointer.
+    auto result = struct_array.View(struct_array.type());
+    TENZIR_ASSERT(result.ok(), result.status().ToString().c_str());
+    return result.MoveValueUnsafe();
```

Comment:
I'm using the pattern `std::make_shared<arrow::StructArray>(struct_array.data())` for this. Am I missing something or is my one valid too? (Not that it matters here)

Reply:
Your approach is strictly better. Thanks!